File: LC058LengthOfLastWord.java
package leetcode;

/**
 * https://leetcode.com/problems/length-of-last-word/
 */
public class LC058LengthOfLastWord {
    public int lengthOfLastWord(String s) {
        if (s == null || s.length() == 0) return 0;
        int n = s.length();
        for (; n > 0 && s.charAt(n - 1) == ' '; n--) ;
        for (int i = n - 1; i >= 0; i--) {
            if (s.charAt(i) == ' ') return n - i - 1;
        }
        return n;
    }

    public static void main(String[] arguments) {
        LC058LengthOfLastWord solution = new LC058LengthOfLastWord();
        System.out.println(solution.lengthOfLastWord("Hello World"));
    }
}

File: algorithms/code/leetcode/lc058_length_of_last_word/lc058_length_of_last_word.py
# https://leetcode.com/problems/length-of-last-word/

import unittest


class Solution:
    def lengthOfLastWord(self, s: str | None) -> int:
        if not s or len(s) == 0:
            return 0
        n = len(s)
        while n > 0 and s[n - 1] == ' ':
            n -= 1
        for i in range(n - 1, -1, -1):
            if s[i] == ' ':
                return n - i - 1
        return n


class TestCode(unittest.TestCase):
    def test_hello_horld(self) -> None:
        assert Solution().lengthOfLastWord('Hello World') == 5

    def test_nothing(self) -> None:
        assert Solution().lengthOfLastWord(None) == 0
        assert Solution().lengthOfLastWord('') == 0

    def test_almost_nothing(self) -> None:
        assert Solution().lengthOfLastWord(' ') == 0

    def test_trailing_space(self) -> None:
        assert Solution().lengthOfLastWord('Hello World  ') == 5

    def test_single_word(self) -> None:
        assert Solution().lengthOfLastWord('HelloWorld') == 10
